File: PRACTICAS/PRACTICA_ACTUAL/JAVA/JAVA/Civitas_P1/src/civitas_p1/SorpresaJugadorEspeculador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas_p1;

import java.util.ArrayList;

/**
 *
 * @author angel
 */
public class SorpresaJugadorEspeculador extends Sorpresa{
    
    private float fianza;
    
    
    SorpresaJugadorEspeculador(String texto, float fianza){
        
        super(texto);
        this.fianza = fianza;
        
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {  
            Jugador player = todos.get(actual);
            todos.set(actual,new JugadorEspeculador(player, fianza));      
    }
    
   
    
}

File: PRACTICAS/PRACTICA_ACTUAL/RUBY/lib/sorpresa_jugadorespeculador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'jugador_especulador.rb'
require_relative 'jugador.rb'

module Civitas
  class Sorpresa_jugadorespeculador < Sorpresa
    
    def initialize(cantidad,t)
      super(t)
      @fianza = cantidad     
      
    end
    
        public_class_method :new

    def aplicar_a_jugador(actual,todos)
      if actual>=0&&actual<todos.size
        informe(actual,todos)
        jugadorespeculador = Jugador_especulador.copia(todos[actual],@fianza)
        todos[actual]=jugadorespeculador
        #todos.push(jugadorespeculador)
        #todos.pop(todos[actual])
        
      end
    end
  end
end
